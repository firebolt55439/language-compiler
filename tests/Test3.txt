/**
* This file tests:
* main function
* 'let' int declaration
* returning from main
* typedefs
* extern non-variadic functions
* output to screen
* extern variadic functions
*/

typedef int : i32;

extern "C" fn putchar:int takes [int test];
extern "C" fn printf takes [i8* fmt, ...];

fn main:i32 takes [int argc, i8** argv]{
	let int test = argc;
	putchar(32);
	printf("args: %d\n", test);
	test = 3;test=4;test=argc+1;
	return test;
}