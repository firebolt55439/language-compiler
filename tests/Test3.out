1 
 args: 1
